File: 136.single-number.cpp
/*
 * @lc app=leetcode id=136 lang=cpp
 *
 * [136] Single Number
 */
#include <map>
#include <vector>
using namespace std;

class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int result = 0;
        for(int i = 0;i < nums.size();++i)
            result ^= nums[i];
        return result;
    }
};

File: code/136.single-number.py
#
# @lc app=leetcode id=136 lang=python3
#
# [136] Single Number
#

# @lc code=start
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
        result = 0
        for num in nums:
            result = result ^ num
        return result

# @lc code=end

